packages/@aws-cdk/aws-apprunner-alpha/lib/vpc-ingress-connection.ts

```diff
@@ -0,0 +1,168 @@
+   */
+  readonly vpcIngressConnectionName: string;
+
+  /**
+   * The domain name associated with the VPC Ingress Connection resource.
+   */
+  readonly domainName: string;
+
+  /**
+   * The current status of the VPC Ingress Connection.
+   */
+  readonly status: string;
+}
+
+/**
+ * Represents the App Runner VPC Ingress Connection.
+ */
+export interface IVpcIngressConnection extends cdk.IResource {
+  /**
+   * The Amazon Resource Name (ARN) of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionArn: string;
+
+  /**
+    * The name of the VPC Ingress Connection.
+    * @attribute
+    */
+  readonly vpcIngressConnectionName: string;
+}
+
+/**
+ * The App Runner VPC Ingress Connection
+ *
+ * @resource AWS::AppRunner::VpcIngressConnection
+ */
+export class VpcIngressConnection extends cdk.Resource implements IVpcIngressConnection {
+  /**
+   * Import from VPC Ingress Connection from attributes.
+   */
+  public static fromVpcIngressConnectionAttributes(scope: Construct, id: string, attrs: VpcIngressConnectionAttributes): IVpcIngressConnection {
+    const vpcIngressConnectionArn = attrs.vpcIngressConnectionArn;
+    const domainName = attrs.domainName;
+    const status = attrs.status;
+    const vpcIngressConnectionName = attrs.vpcIngressConnectionName;
+
+    class Import extends cdk.Resource implements IVpcIngressConnection {
+      public readonly vpcIngressConnectionArn = vpcIngressConnectionArn;
+      public readonly domainName = domainName;
+      public readonly status = status;
+      public readonly vpcIngressConnectionName = vpcIngressConnectionName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner VPC Ingress Connection from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, vpcIngressConnectionArn: string): IVpcIngressConnection {
+    const resourceParts = cdk.Fn.split('/', vpcIngressConnectionArn);
+
+    const vpcIngressConnectionName = cdk.Fn.select(0, resourceParts);
+
+    class Import extends cdk.Resource implements IVpcIngressConnection {
+      public readonly vpcIngressConnectionName = vpcIngressConnectionName;
+      public readonly vpcIngressConnectionArn = vpcIngressConnectionArn;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionArn: string;
+
+  /**
+   * The domain name associated with the VPC Ingress Connection resource.
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The current status of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly status: string;
+
+  /**
+   * The name of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionName: string;
+
+  public constructor(scope: Construct, id: string, props: VpcIngressConnectionProps) {
+    super(scope, id, {
+      physicalName: props.vpcIngressConnectionName,
```

Comment:
Did you check if it deploys with `vpcIngressConnectionName: undefined`?
If not, we should lazily generate one.

Reply:
Thank you. I tried deploying with `vpcIngressConnectionName: undefined`, and I was able to deploy without any issues.

<img width="544" alt="image" src="https://github.com/user-attachments/assets/58a1be36-06c7-4b5f-9a11-90055d7852af">

<img width="624" alt="image" src="https://github.com/user-attachments/assets/1caaeaf7-2568-49fb-9396-b20cd6ad6803">